CenterForOpenScience/osf.io
# [ENG-897] EGAP Ingester

## Purpose

Turn those wonderful local file structures in to EGAP projects!

## Changes

Adds a management command that creates a node with proper contributor and uploads files maintaining directory hierarchy with tests.

## QA Notes
To setup this you must:
1) log in and create a private project
2) upload a zip of file's using the EGAP import structure to the project
3) run `python manage.py import_EGAP -c=<username> -id=<guid>`

- This is a low risk change that doesn't involve a migration. 
- QA will be able to verify this with an Ipython notebook. Once that's set-up they should be able to browse the registrations and see they were imported properly with the correct permissions and bibliographic status.
- This is unlikely to effect any other portion of the site in a substantive way.

This includes some unittests but obviously can easily be snagged on irregular data, if you find any please report it to me.

## Documentation

Code comments, JIRA

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-897
